Cmtool usage updates

cmtool opened the pull request.

cmtool commented:
This is the first time I've ever attempted to do a pull request on GitHub, so my apologies if I've messed this up. This is basically a bunch of spelling and grammar edits, along with changes to community.md that involve moving things that have been archived or depreciated to the right sections, fixing links.

The one area where feedback would be appreciated is the move the listservs to 'falling out of use'. I went through some of the archives, and it seems like discussion on the listservs is very slow, but I'd really welcome some feedback as to whether it's fine to label listservs as slowly being replaced by mattermost.

cmtool commented:
Just saw some errors, so I closed this PR!

mlutfy commented on docs/api/usage.md:
```diff
@@ -51,8 +51,7 @@ array(
 
-!!! note
-    A few specialized actions like `getsingle` or `getvalue` may return success in a different format.
+*Note*: A few specialized actions like `getsingle` or `getvalue` may return success in a different format.
```
I think the `!!! note` seems OK?

![Capture d’écran de 2019-05-09 15-13-23](https://user-images.githubusercontent.com/254741/57480213-158bdc80-726d-11e9-8f06-a429eefe186c.png)

cmtool replied:
```suggestion
!!! note
A few specialized actions like `getsingle` or `getvalue` may return success in a different format.
```

mlutfy commented:
@cmtool It seemed OK to me? There was just the `!!! note` that I would personally have left as-is. I think you can continue to edit changes on a PR, instead of opening a new one?

Random example, if go to "Changes", then next to the file header:

![Capture d’écran de 2019-05-09 15-17-08](https://user-images.githubusercontent.com/254741/57480380-8a5f1680-726d-11e9-89dd-34862b04c3ea.png)

cmtool commented:
@mlutfy 

> @cmtool It seemed OK to me? There was just the `!!! note` that I would personally have left as-is. I think you can continue to edit changes on a PR, instead of opening a new one?

Haha, that's exactly the thing I was like 'oh no, it actually generates a thing that's really useful, oops'. XD

>Random example, if go to "Changes", then next to the file header:
Ah, okay, will do that! Thank you! :D

cmtool commented on docs/api/usage.md:
```diff
@@ -52,7 +52,7 @@ array(
 
 !!! note
-    A few specialized actions like `getsingle` or `getvalue` may return success in a different format.
+A few specialized actions like `getsingle` or `getvalue` may return success in a different format.
```
```suggestion
    A few specialized actions like `getsingle` or `getvalue` may return success in a different format.
```

cmtool commented:
@mlutfy Okay, I went in and did two man changes. 

1. I added back in the !!! note
2. I moved IRC from ## Falling out of use to ## Out of use, based on the Mattermost documentation channel discussion where it was mentioned that CiviCRM no longer officially supports that channel.

Let me know if there are any further changes/edits you'd like me to make!
